core/src/main/java/org/elasticsearch/search/internal/InternalSearchHit.java

```diff
@@ -808,7 +810,9 @@ public void writeTo(StreamOutput out, InternalSearchHits.StreamContext context)
             out.writeVInt(innerHits.size());
             for (Map.Entry<String, InternalSearchHits> entry : innerHits.entrySet()) {
                 out.writeString(entry.getKey());
-                entry.getValue().writeTo(out, InternalSearchHits.streamContext().streamShardTarget(InternalSearchHits.StreamContext.ShardTargetType.NO_STREAM));
+                ShardTargetType previous = InternalSearchHits.streamContext().streamShardTarget();
```

Comment:
it is the `ShardTargetType` of the current hit and we remember that so that it doesn't get lost when the inner_hits are serialised. Because for the inner hits we don't need shard target, because that is always the same as the top level search hit (shard and index that is).

I'm ok with renaming it to `shardTarget`.